src/TaskCreationSection/CategoryCreationSection/AddCategoryModal.jsx

```diff
@@ -76,67 +66,64 @@ export default function AddCategoryModal({ addCategory }) {
       });
       return;
     }
-    if (newCategory.name !== "" && isCategoryAlreadyAdded()) {
-      addCategory({
-        name: newCategory.name.toLowerCase(),
-        icon: newCategory.icon,
-        colorStyle: newCategory.colorStyle,
-        isAddedByUser: true,
-        uuid: uuid(),
-      });
+    const isCategoryValid =
+      newCategory.name &&
+      !categories.some(({ name }) => newCategory.name === name);
 
-      setNewCategory({ ...newCategory, name: "" });
-      showAlert({
-        title: "New category has been added!",
-        type: "success",
-        background: "bg-success",
-        isShowed: true,
-      });
-    } else {
+    if (!isCategoryValid) {
       setIsCorrectTyped(false);
       showAlert({
         title: "You can't create a category with this name, try again!",
         type: "error",
         background: "bg-error",
         isShowed: true,
       });
+      return;
     }
+
+    addCategory({ ...newCategory, uuid: uuid() });
+    showAlert({
+      title: "New category has been added!",
+      type: "success",
+      background: "bg-success",
+      isShowed: true,
+    });
   };
 
   const iconPicker = useMemo(
```

Comment:
next task can be to get rid of these usages `useMemo` for the internal component, it seems to be redundant. If rendering this component is costly and you really want to avoid it, then move it to separate component like
```
const IconPicker = Reac.memo(({ ...props }) => {
...
})
````